lib/Driver/ToolChain.cpp

```diff
@@ -319,7 +310,9 @@ ToolChain::constructBatchJob(ArrayRef<const Job *> jobs,
   if (jobs.empty())
     return nullptr;
 
-  assert(jobsAreSubsequenceOfCompilationInputs(jobs, C));
+  llvm::SmallVector<const Job *, 16> sortedJobs;
+  sortJobsToMatchCompilationInputs(jobs, sortedJobs, C);
+  jobs = sortedJobs;
```

Comment:
@davidungar would probably prefer you just use `sortedJobs` everywhere else; if you want to keep the diff small, you could rename the parameter to `unsortedJobs` instead.

Reply:
Yes, @jrose-apple , you read my mind!